### ISSUE TITLE
Clean up abandoned, non-functional Sentry code 

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#3079]
> **Note**: This issue has been created to document an old WIP pull request that had become stale. No issue had been opened for that PR, so this issue is to discuss whether it's worth starting fresh and continuing that work.

Before it was closed, PR #2552 aimed to update how Sentry was used in SCT. Are the fixes from #2552 still needed?

**Comment by jcohenadad** [comment]
> Are the fixes from #2552 still needed?

I have no idea. Maybe @zougloub has better insights as he worked on this

**Comment by joshuacwnewton** [comment]
Last year, I dug into this on slack, but I didn't find the time to summarize the current state of Sentry on the GitHub issue. So, I'm doing that now!

TL;DR:

> It looks like all of the code necessary to perform Sentry logging is present: Our sentry usage is confined entirely to functions within [`sys.py`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/utils/sys.py). Sentry is initialized by the [`init_error_client`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2d09af3ee707ca166c3d1611295a7575ffdd4cd0/spinalcordtoolbox/utils/sys.py#L231-L260) function, which is called by the [`init_sct`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2d09af3ee707ca166c3d1611295a7575ffdd4cd0/spinalcordtoolbox/utils/sys.py#L190-L193) function that gets called at the start of every CLI script. 
> 
> However, initialization is dependent on the [`SENTRY_DSN`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2d09af3ee707ca166c3d1611295a7575ffdd4cd0/spinalcordtoolbox/utils/sys.py#L232) environment variable being set. Currently, we do not set this variable, and because we never really settled on a way to set `SENTRY_DSN`, the code has laid dormant for years.

On `SENTRY_DSN`:

>  In the `init_error_client` function, there's a clue about how `SENTRY_DSN` used to be set here:
>
> https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2d09af3ee707ca166c3d1611295a7575ffdd4cd0/spinalcordtoolbox/utils/sys.py#L256
>
> So, it looks like we used to set `SENTRY_DSN` in our launcher, but... we do not have an `sct_launcher` binary anymore! We do have [`launcher.py`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2d09af3ee707ca166c3d1611295a7575ffdd4cd0/spinalcordtoolbox/compat/launcher.py#L20-L26), which is where we set our environment variables for CLI scripts, but `SENTRY_DSN` is not present there.

On `sentry_dsn.py`:

> Additionally, `install_sct` contains the following lines:
>
> https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2d09af3ee707ca166c3d1611295a7575ffdd4cd0/install_sct#L431-L447
>
> But I've never actually seen this `sentry_dsn.py` file get created, nor does it seem like there is any part of SCT's codebase that tries to actually run `sentry_dsn.py`.

On current Sentry activity:

> The Sentry login information is currently stored in `pass`. Logging into the web portal, we see the following:
>
> ![image](https://user-images.githubusercontent.com/16181459/215205243-5f8bdd75-5e42-47af-b294-8d78dd000351.png)
> 
> There's an old (v4.0.0-beta-1) installation of SCT out there somewhere, still sending error reports... We got a ping as of 1wk ago! But, that's it? These three reports are the only issues in our sentry account over the last 30 days. (Mostly because logging after 90 days is a paid service, but likely also because Sentry is only working on older installations of SCT.)

So, tl;dr: Sentry used to work; it no longer works; it could potentially be re-enabled by changing how we set the DSN.
